homeassistant/components/kaiterra/sensor.py

```diff
@@ -0,0 +1,110 @@
+"""Support for Kaiterra Temperature ahn Humidity Sensors."""
+from homeassistant.helpers.entity import Entity
+
+from homeassistant.helpers.dispatcher import async_dispatcher_connect
+
+from homeassistant.const import CONF_DEVICE_ID, CONF_NAME
+
+from .const import DOMAIN, DISPATCHER_KAITERRA
+
+SENSORS = [
+    {
+        "name": "Temperature",
+        "prop": "rtemp",
+        "icon": "mdi:temperature-celsius",
+        "device_class": "temperature",
+    },
+    {
+        "name": "Humidity",
+        "prop": "rhumid",
+        "icon": "mdi:water",
+        "device_class": "humidity",
+    },
+]
+
+
+async def async_setup_platform(hass, config, async_add_entities, discovery_info=None):
+    """Set up the kaiterra temperature and humidity sensor."""
+    if discovery_info is None:
+        return
+
+    api = hass.data.get(DOMAIN)
+    name = discovery_info.get(CONF_NAME)
+    device_id = discovery_info.get(CONF_DEVICE_ID)
+
+    async_add_entities(
+        [KaiterraSensor(api, name, device_id, sensor) for sensor in SENSORS]
+    )
+
+
+class KaiterraSensor(Entity):
+    """Implementation of a Kaittera sensor."""
+
+    def __init__(self, api, name, device_id, sensor):
+        """Initialize the sensor."""
+        self._api = api
+        self._name = f'{name} - {sensor.get("name")}'
+        self._device_id = device_id
+        self._kind = sensor.get("name").lower()
+        self._icon = sensor.get("icon")
+        self._property = sensor.get("prop")
+        self._device_class = sensor.get("device_class")
+
+    @property
+    def _sensor(self):
+        """Return the sensor data."""
+        return self._api.data.get(self._device_id, {}).get(self._property, {})
+
+    @property
+    def should_poll(self):
+        """Return that the sensor should not be polled."""
+        return False
+
+    @property
+    def available(self):
+        """Return the availability of the sensor."""
+        return self._api.data.get(self._device_id) is not None
+
+    @property
+    def device_class(self):
+        """Return the device class."""
+        return self._device_class
+
+    @property
+    def name(self):
+        """Return the name."""
+        return self._name
+
+    @property
+    def icon(self):
+        """Return the icon."""
+        return self._icon
```

Comment:
Sensor entities with device class already shows the correct icon in the frontend.